### PR TITLE
[FIXED JENKINS-34789] Release builds should not trigger downstream projects

### PR DESCRIPTION
**Pull request by fbelzunc**
screen shot 2016-05-13 at 11 34 11

https://issues.jenkins-ci.org/browse/JENKINS-34789

@reviewbybees @stephenc @aheritier

### CODE REVIEW

**Comment by aheritier** [comment]
cc @olamy

**Comment by fbelzunc** [comment]
@reviewbybees Hello busy bees, can I get a review here, please?

**Comment by aheritier** [comment]
@fbelzunc LGTM 🐝

**Comment by aheritier** [comment]
@fbelzunc cannot you rebase your PR on current master. All tests should be good now.

File: src/main/java/hudson/maven/MavenModuleSetBuild.java [hunk 1]
```diff
@@ -1438,6 +1438,23 @@ private String getRootPath(String prefix) {
...
     }
+
+    /**
+     * Check if the current build is a release or not.
+     * If all its maven modules are not SNAPSHOT then it is a release
+     *
+     * @since 2.12.2
+     * @return true if {@link MavenModuleSetBuild} is a release
+     */
+    protected Boolean isARelease() {
```
Comment (KostyaSha) [inline src/main/java/hudson/maven/MavenModuleSetBuild.java]: maybe `isRelease()`?

**Comment by KostyaSha** [comment]
@aheritier rebase and merge button now available in UI

**Comment by fbelzunc** [comment]
@aheritier The build is successful now and I think I addressed the comments from @KostyaSha 

BTW @KostyaSha thanks for your time reviewing.

**Comment by aheritier** [comment]
LGTM @fbelzunc I'll wait that @KostyaSha validates that your change is ok and I'll merge

File: src/main/java/hudson/maven/MavenModuleSetBuild.java [hunk 2]
```diff
@@ -1438,6 +1438,23 @@ private String getRootPath(String prefix) {
...
+     * Check if the current build is a release or not.
+     * If all its maven modules are not SNAPSHOT then it is a release
+     *
+     * @since 2.12.2
+     * @return true if {@link MavenModuleSetBuild} is a release
+     */
+    protected Boolean isRelease() {
+        MavenModuleSet mavenModuleSet = this.getProject();
+        for (MavenModule mavenModule : mavenModuleSet.getModules()) {
+            if(mavenModule.getVersion().contains("-SNAPSHOT")) {
```
Comment (olamy) [inline src/main/java/hudson/maven/MavenModuleSetBuild.java]: why not using https://maven.apache.org/ref/3.3.9/maven-artifact/apidocs/org/apache/maven/artifact/ArtifactUtils.html there is a method 	isSnapshot(String version)

Comment (aheritier) [inline src/main/java/hudson/maven/MavenModuleSetBuild.java]: @fbelzunc could you try to address this remark this week ? I would like to merge your PR and release the plugin this week. Thx

**Pull request merged**